packages/core/src/Page/index.ts

```diff
@@ -443,7 +440,8 @@ export class Page {
    */
   buildPageNav(content: string) {
     const isFmPageNavSpecifierValid = this.isPageNavigationSpecifierValid();
-    const doesLayoutHavePageNav = this.pageConfig.layoutManager.layoutHasPageNav(this.layout);
+    const doesLayoutHavePageNav = this.layout ? this.pageConfig.layoutManager.layoutHasPageNav(
+      this.layout) : false;
```

Comment:
I think we can avoid all of these typing issues with `this.layout`, and the additions of `?? LAYOUT_DEFAULT_NAME` below, by asserting that the `layout` attribute exists with a `!` operator. 
You'll need to modify [this](https://github.com/ong6/markbind/blob/f11595a93c3cd3dea8f815606875a9498461062a/packages/core/src/Page/PageConfig.ts#L79) too (just delete the question mark)

Feel free to trace to verify that this is safe!